mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -742,18 +742,45 @@ fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
     }
 }
 
-pub fn dex_fee_amount(base: &str, rel: &str, trade_amount: &MmNumber, min_tx_amount: &MmNumber) -> MmNumber {
+pub fn dex_fee_amount(base: &str, rel: &str, trade_amount: &MmNumber, min_tx_amount: &MmNumber) -> DexFee {
     let rate = dex_fee_rate(base, rel);
-    let fee_amount = trade_amount * &rate;
-    if &fee_amount < min_tx_amount {
-        min_tx_amount.clone()
-    } else {
-        fee_amount
+    let fee = trade_amount * &rate;
+
+    if &fee <= min_tx_amount {
+        return DexFee::Standard(min_tx_amount.clone());
+    }
+
+    if base == "KMD" {
+        // Drop the fee by 25%, which will be burned during the taker fee payment.
+        //
+        // This cut will be dropped before return if the final amount is less than
+        // the minimum transaction amount.
+
+        // Fee with 25% cut
+        let new_fee = &fee * &MmNumber::from("0.75");
+
+        let (fee, burn) = if &new_fee >= min_tx_amount {
+            // Use the max burn value, which is 25%.
+            let burn_amount = &fee - &new_fee;
+
+            (new_fee, burn_amount)
+        } else {
+            // Burn only the exceed amount because fee after 25% cut is less
+            // than `min_tx_amount`.
+            let burn_amount = &fee - min_tx_amount;
+            let new_fee = &fee - &burn_amount;
```

Comment:
No need for `let new_fee = &fee - &burn_amount;` we can just return `min_tx_amount` as the `new_fee`.